You'll need this patch for 2.6.26 to replace the changes in mtd_blkdevs.c.
Yes, I know this patch is really bad, but I can't see a nice way to do this, and I 
need to specify the device name as /dev/mtdblock:cramfs on the kernel 
command line. Once the fab4 2.6.26 kernel boots further you'll hit this 
problem.

Cheers,
Richard
---
 drivers/mtd/mtd_blkdevs.c |    5 +++++
 1 file changed, 5 insertions(+)

Index: linux-2.6.26/drivers/mtd/mtd_blkdevs.c
===================================================================
--- linux-2.6.26.orig/drivers/mtd/mtd_blkdevs.c	2009-03-28 16:50:00.000000000 +0100
+++ linux-2.6.26/drivers/mtd/mtd_blkdevs.c	2009-03-28 16:51:20.000000000 +0100
@@ -276,6 +276,11 @@ int add_mtd_blktrans_dev(struct mtd_blkt
 		snprintf(gd->disk_name, sizeof(gd->disk_name),
 			 "%s%d", tr->name, new->devnum);
 
+	/* HACK: rename the device to allow it to be mounted at boot time
+	 * using /dev/mtdblock:name */
+	snprintf(gd->disk_name, sizeof(gd->disk_name),
+		 "%s:%s", tr->name, new->mtd->name);
+
 	/* 2.5 has capacity in units of 512 bytes while still
 	   having BLOCK_SIZE_BITS set to 10. Just to keep us amused. */
 	set_capacity(gd, (new->size * tr->blksize) >> 9);
